sip-test-kit/src/main/java/de/ikor/sip/foundation/testkit/workflow/whenphase/routeinvoker/impl/FileRouteInvoker.java

```diff
@@ -0,0 +1,92 @@
+package de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.impl;
+
+import static de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.headers.FileExchangeHeaders.*;
+
+import de.ikor.sip.foundation.testkit.util.SIPExchangeHelper;
+import de.ikor.sip.foundation.testkit.workflow.givenphase.Mock;
+import de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.RouteInvoker;
+import java.util.Map;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.*;
+import org.apache.camel.component.file.FileConsumer;
+import org.apache.camel.component.file.FileEndpoint;
+import org.apache.camel.support.EmptyAsyncCallback;
+import org.springframework.stereotype.Component;
+
+/** Invoker class for triggering routes with File consumer */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class FileRouteInvoker implements RouteInvoker {
+
+  private final CamelContext camelContext;
+
+  @Override
+  public Exchange invoke(Exchange inputExchange, Endpoint endpoint) {
+    Route route =
+        camelContext.getRoute(
+            (String) inputExchange.getProperty(Mock.ENDPOINT_ID_EXCHANGE_PROPERTY));
+    FileConsumer fileConsumer = (FileConsumer) route.getConsumer();
+
+    Exchange fileExchange = createFileExchange(fileConsumer, inputExchange);
+
+    fileConsumer.getAsyncProcessor().process(fileExchange, EmptyAsyncCallback.get());
+
+    return SIPExchangeHelper.createEmptyExchange(camelContext);
```

Comment:
We can convert the interface to return Optional<Exchange>
Than we can return Optional.empty(). It would be easier to understand

Reply:
Refactored to return Optional<Exchange>.